File: templates/nextjs/sentry.config.ts
/**
 * Sentry Configuration — Next.js
 *
 * Initializes Sentry for error monitoring in a Next.js application.
 * This file should be imported in your `instrumentation.ts` (App Router)
 * or `_app.tsx` (Pages Router).
 *
 * Setup:
 *   1. npm install @sentry/nextjs
 *   2. Add NEXT_PUBLIC_SENTRY_DSN to your .env file
 *   3. Run: npx @sentry/wizard@latest -i nextjs
 *      (or manually configure using this template)
 *
 * Environment Variables:
 *   NEXT_PUBLIC_SENTRY_DSN — Your Sentry project DSN (required)
 *   SENTRY_AUTH_TOKEN      — For source map uploads in CI (optional)
 *
 * See: https://docs.sentry.io/platforms/javascript/guides/nextjs/
 */

import * as Sentry from "@sentry/nextjs";

Sentry.init({
  // DSN is read from environment variable — never hardcode this
  dsn: process.env.NEXT_PUBLIC_SENTRY_DSN,

  // Performance monitoring — adjust sample rate for production traffic
  // 1.0 = 100% of transactions captured (reduce for high-traffic apps)
  tracesSampleRate: process.env.NODE_ENV === "production" ? 0.2 : 1.0,

  // Session replay — captures user interactions leading up to errors
  // Only available on Sentry's paid plans; safe to leave enabled on free tier (just won't record)
  replaysSessionSampleRate: 0.1,
  replaysOnErrorSampleRate: 1.0,

  // Only send errors in production (avoid noise during development)
  enabled: process.env.NODE_ENV === "production",

  // Filter out known non-actionable errors
  ignoreErrors: [
    // Browser extensions and third-party scripts
    "ResizeObserver loop",
    "Non-Error promise rejection captured",
    // Network errors that are expected
    "Failed to fetch",
    "NetworkError",
    "Load failed",
  ],

  // Set the environment tag for filtering in Sentry dashboard
  environment: process.env.NODE_ENV,
});

File: templates/flutter/sentry-config.dart
/// Sentry Configuration — Flutter
///
/// Initializes Sentry for error monitoring in a Flutter application.
///
/// Setup:
///   1. Add to pubspec.yaml: sentry_flutter: ^8.0.0
///   2. Add SENTRY_DSN to your .env file (use flutter_dotenv or --dart-define)
///   3. Replace your main() function:
///      ```dart
///      Future<void> main() async {
///        await initSentry(() => runApp(const MyApp()));
///      }
///      ```
///
/// Environment Variables:
///   SENTRY_DSN — Your Sentry project DSN (required)
///
/// See: https://docs.sentry.io/platforms/flutter/

import 'package:flutter/foundation.dart';
import 'package:sentry_flutter/sentry_flutter.dart';

/// Initialize Sentry and run the app inside Sentry's error boundary.
///
/// [appRunner] is the function that starts your app (e.g., `() => runApp(MyApp())`).
/// Sentry wraps this to catch any unhandled errors during startup.
Future<void> initSentry(Future<void> Function() appRunner) async {
  await SentryFlutter.init(
    (options) {
      // DSN is read from environment — pass via --dart-define=SENTRY_DSN=...
      // or use flutter_dotenv to load from .env file
      options.dsn = const String.fromEnvironment('SENTRY_DSN');

      // Performance monitoring — adjust for production traffic
      options.tracesSampleRate = kReleaseMode ? 0.2 : 1.0;

      // Only send errors in release builds
      options.enabled = kReleaseMode;

      // Capture failed HTTP requests automatically
      options.enableAutoPerformanceTracing = true;

      // Attach device info to error reports
      options.attachThreads = true;

      // Set environment tag
      options.environment = kReleaseMode ? 'production' : 'development';
    },
    appRunner: appRunner,
  );
}
